Automattic/wp-calypso
# Offer Reset: Redirect to WPCOM when in Calypso Green

#### Changes proposed in this Pull Request

* Since the Plans grid is also used in Calypso Green, we need to make sure that in that case, we redirect users to WPCOM when they select a product since neither the checkout URL nor the Jetpack Connect flow exist in Calypso Green.
* Removed unused placeholder component.

#### Testing instructions

* Run this PR with `yarn start-jetpack-cloud`.
* Visit `cloud.jetpack.com/pricing`.
* Go to the checkout with any product.
* Make sure you were redirected to WordPress.com.
* Now you might have to start the Calypso build process or use the test URL.
* Make sure the Plans page still works fine in WordPress.com (`/plans/:site`).
* Make sure the Plans page still works fine in the Connect Flow WordPress.com (`/jetpack/connect`).

Fixes 1169247016322522-as-1193016541209410

#### Demo
![Kapture 2020-09-10 at 10 53 20](https://user-images.githubusercontent.com/3418513/92740161-e64f1900-f353-11ea-9af0-dbee0d259a15.gif)
